### PR TITLE
[ISSUES#115]: Enable play with 2 players

### LINKED ISSUE #115: Ludo contracts: Enable play with 2 players
## Issue Overview

The game session currently requires waiting for all 4 players to join before starting. This creates unnecessary delays when fewer players want to begin playing immediately.

## Proposed Solutions or Ideas

- Enable games to start with a minimum of 2 players present o max 4 players, `No` option for 3 players for now
- Check all edge cases
- Implement and update test for this new feature

Check the figma [design](https://www.figma.com/design/vwxsBUQOK2nKsmzUEDs4xF/Public-Marquis?node-id=19-493&p=f&t=2n3Ivp4QE7oSDdvv-0) as reference 

image

### PR DESCRIPTION
**Pull request by 847850277**
# Task name here

Fixes #115 

@gianalarcon PTAL.When you are free.

## Types of change

- [ ] Feature
- [ ] Bug
- [ ] Enhancement

### CODE REVIEW

File: packages/snfoundry/contracts/src/test/TestContract.cairo [hunk 1]
```diff
@@ -339,7 +369,8 @@ fn should_create_new_game_session() {
     let marquis_game_dispatcher = IMarquisGameDispatcher { contract_address: ludo_contract };
     let token = ZERO_TOKEN();
     let amount = 0;
-    let session_id = marquis_game_dispatcher.create_session(token, amount);
+    let two_required_players = 2;
```
Comment (gianalarcon) [inline packages/snfoundry/contracts/src/test/TestContract.cairo:372]: here we should have two test, `should_create_new_game_session_with_2_players` and `should_create_new_game_session_with_4_players`. Almost same requirement for the other tests

**Review by gianalarcon — CHANGES_REQUESTED** [review]

**Review by gianalarcon — COMMENTED** [review]
Looks good, almost done! 
Lets organize much better the tests folder, we can have 4 files
1. setup
2. Pregame tests or general tests
3. Session_with_two_players
4. Session_with_four_players

**Comment by 847850277** [comment]
> Looks good, almost done! Lets organize much better the tests folder, we can have 4 files
> 
> 1. setup
> 2. Pregame tests or general tests
> 3. Session_with_two_players
> 4. Session_with_four_players

fixed.
